From: Michael Olbrich <m.olbrich@pengutronix.de>
Subject: make libparted dependency optional

This patch adds a configure option for libparted.
When disabled, the helper programs using it will not be built.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

---
 configure.ac    |   11 ++++++++---
 src/Makefile.am |   18 +++++++++++++-----
 2 files changed, 21 insertions(+), 8 deletions(-)

Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -151,9 +151,14 @@ PKG_CHECK_MODULES(POLKIT_BACKEND_1, [pol
 AC_SUBST(POLKIT_BACKEND_1_CFLAGS)
 AC_SUBST(POLKIT_BACKEND_1_LIBS)
 
-PKG_CHECK_MODULES(LIBPARTED, [libparted >= 1.8.8])
-AC_SUBST(LIBPARTED_CFLAGS)
-AC_SUBST(LIBPARTED_LIBS)
+AC_ARG_ENABLE(libparted,[  --disable-libparted],enable_libparted=$enableval,enable_libparted=yes)
+
+if test "x$enable_libparted" = xyes; then
+  PKG_CHECK_MODULES(LIBPARTED, [libparted >= 1.8.8])
+  AC_SUBST(LIBPARTED_CFLAGS)
+  AC_SUBST(LIBPARTED_LIBS)
+fi
+AM_CONDITIONAL(LIBPARTED_ENABLED, test "x$enable_libparted" = xyes)
 
 PKG_CHECK_MODULES(DEVMAPPER, [devmapper >= 1.02])
 AC_SUBST(DEVMAPPER_CFLAGS)
Index: b/src/Makefile.am
===================================================================
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -70,16 +70,14 @@ devkit_disks_daemon_LDADD = 				\
 	$(LIBATASMART_LIBS)				\
 	$(NULL)
 
+if LIBPARTED_ENABLED
 noinst_LTLIBRARIES = libpartutil.la
 libpartutil_la_SOURCES = partutil.h partutil.c
 libpartutil_la_CPPFLAGS = $(LIBPARTED_CFLAGS)
 libpartutil_la_LIBADD = $(LIBPARTED_LIBS)
+endif
 
 libexec_PROGRAMS += devkit-disks-helper-mkfs                    	\
-		    devkit-disks-helper-delete-partition        	\
-		    devkit-disks-helper-create-partition        	\
-		    devkit-disks-helper-modify-partition        	\
-		    devkit-disks-helper-create-partition-table		\
 		    devkit-disks-helper-change-filesystem-label 	\
 		    devkit-disks-helper-linux-md-remove-component	\
 		    devkit-disks-helper-fstab-mounter			\
@@ -99,6 +97,12 @@ devkit_disks_helper_mkfs_SOURCES = job-s
 devkit_disks_helper_mkfs_CPPFLAGS = $(AM_CPPFLAGS)
 devkit_disks_helper_mkfs_LDADD = $(GLIB_LIBS)
 
+if LIBPARTED_ENABLED
+libexec_PROGRAMS += devkit-disks-helper-delete-partition        	\
+		    devkit-disks-helper-create-partition        	\
+		    devkit-disks-helper-modify-partition        	\
+		    devkit-disks-helper-create-partition-table		\
+		    $(NULL)
 devkit_disks_helper_delete_partition_SOURCES = job-shared.h job-delete-partition.c
 devkit_disks_helper_delete_partition_CPPFLAGS = $(AM_CPPFLAGS)
 devkit_disks_helper_delete_partition_LDADD = $(GLIB_LIBS) libpartutil.la
@@ -114,6 +118,7 @@ devkit_disks_helper_modify_partition_LDA
 devkit_disks_helper_create_partition_table_SOURCES = job-shared.h job-create-partition-table.c
 devkit_disks_helper_create_partition_table_CPPFLAGS = $(AM_CPPFLAGS)
 devkit_disks_helper_create_partition_table_LDADD = $(GLIB_LIBS) libpartutil.la
+endif
 
 devkit_disks_helper_change_filesystem_label_SOURCES = job-shared.h job-change-filesystem-label.c
 devkit_disks_helper_change_filesystem_label_CPPFLAGS = $(AM_CPPFLAGS)
@@ -149,11 +154,14 @@ devkit_disks_helper_drive_poll_LDADD =  
 
 # TODO: move to udev
 udevhelperdir = $(slashlibdir)/udev
-udevhelper_PROGRAMS = devkit-disks-part-id devkit-disks-dm-export devkit-disks-probe-ata-smart
+udevhelper_PROGRAMS = devkit-disks-dm-export devkit-disks-probe-ata-smart
 
+if LIBPARTED_ENABLED
+udevhelper_PROGRAMS += devkit-disks-part-id
 devkit_disks_part_id_SOURCES = part-id.c
 devkit_disks_part_id_CPPFLAGS = $(AM_CPPFLAGS) $(LIBUDEV_CFLAGS)
 devkit_disks_part_id_LDADD = $(GLIB_LIBS) $(LIBUDEV_LIBS) libpartutil.la
+endif
 
 devkit_disks_dm_export_SOURCES = devkit-disks-dm-export.c
 devkit_disks_dm_export_CPPFLAGS = $(AM_CPPFLAGS) $(DEVMAPPER_CFLAGS)
